crates/hc_sandbox/src/zome_call.rs

```diff
@@ -250,40 +244,38 @@ pub async fn zome_call(zome_call: ZomeCall, admin_port: Option<u16>) -> anyhow::
 
     let (auth, key) = generate_signing_credentials(passphrase).await?;
 
-    let (nonce, expires_at) = holochain_nonce::fresh_nonce(Timestamp::now())
-        .map_err(|e| anyhow::anyhow!("Failed to generate nonce: {:?}", e))?;
-
-    let params = ZomeCallParams {
-        provenance: AgentPubKey::from_raw_32(key.verifying_key().as_bytes().to_vec()),
-        cell_id: cell_ids.first().unwrap().clone(),
-        zome_name: ZomeName::from(zome_call.zome_name.clone()),
-        fn_name: FunctionName(zome_call.function.clone()),
-        cap_secret: Some(auth.cap_secret),
-        payload: ExternIO::encode(serde_json::from_slice::<serde_json::Value>(
-            zome_call.payload.as_bytes(),
-        )?)?,
-        nonce,
-        expires_at,
-    };
-    let (payload, hash) = params.serialize_and_hash()?;
-
-    let sig = key.try_sign(&hash)?;
-
-    let response = client
-        .request(AppRequest::CallZome(Box::new(ZomeCallParamsSigned {
-            bytes: ExternIO::from(payload),
-            signature: Signature::try_from(sig.to_vec())?,
-        })))
+    let credentials: Vec<(CellId, SigningCredentials)> = cell_ids
+        .clone()
+        .into_iter()
+        .map(|cell_id| {
+            (
+                cell_id,
+                SigningCredentials {
+                    signing_agent_key: AgentPubKey::from_raw_32(
+                        key.verifying_key().as_bytes().to_vec(),
+                    ),
+                    keypair: key.clone(),
+                    cap_secret: auth.cap_secret,
+                },
+            )
+        })
+        .collect();
+
+    let app_client =
+        get_app_client(&admin_client, zome_call.app_id.clone(), Some(credentials)).await?;
+
+    let response = app_client
+        .call_zome(
+            ZomeCallTarget::CellId(cell_ids.first().unwrap().clone()),
+            ZomeName::from(zome_call.zome_name),
+            FunctionName(zome_call.function),
+            ExternIO::encode(serde_json::from_slice::<serde_json::Value>(
+                zome_call.payload.as_bytes(),
+            )?)?,
+        )
         .await?;
 
-    match response {
-        AppResponse::ZomeCalled(response) => {
-            let response: serde_json::Value = response.decode()?;
-            serde_json::to_writer(std::io::stdout(), &response)?;
-            println!(); // Add newline
-        }
-        other => anyhow::bail!("Unexpected response while calling zome: {:?}", other),
-    }
+    println!("{}", response.decode::<serde_json::Value>()?);
```

Comment:
Using a format string looks like it's executed first, before being sent to stdout. I think using stdout was a little more efficient here, rather than going via a string in memory

Reply:
I actually changed that because I didn't get the tests passing with the other approach. The tests claimed that stdout was empty and I didn't understand why...

Reply:
Hmm...maybe the problem was just that I did the decoding directly inside the `serde_json::to_writer()` function. Changed it now: https://github.com/holochain/holochain/pull/4968/commits/d4e781e662c27bd271a6b3369729ddd478af385e

Reply:
Even more compact: https://github.com/holochain/holochain/pull/4968/commits/31372d80b888b305216dae3b2b50cffe8ee033a9

I think I was missing the turbofish `::<serde_json::Value>` when the tests failed initially.

Reply:
Nice! It's possible stdout hadn't been flushed or something like that too? Changes look good though :)